core/isolate.rs

```diff
@@ -364,17 +365,15 @@ impl Isolate {
   pub fn clear_exception(&mut self) {
     let isolate = self.v8_isolate.as_ref().unwrap();
     let mut locker = v8::Locker::new(isolate);
```

Comment:
This unnecessary locker is abstraction induced runtime overhead. If we're unable to remove it, maybe the abstraction is not correct.

Reply:
1. It's not clear to me that the use of Locker here is actually unnecessary.
2. You have a better abstraction in mind? I'm all ears.

Reply:
I can definitely remove the Locker in clear_exception from certain call stacks. But it seems that it also gets called sort of randomly from OP futures?

Reply:
> It's not clear to me that the use of Locker here is actually unnecessary.

It's wasn't necessary in C++
https://github.com/denoland/deno/blob/v0.28.0/core/libdeno/api.cc#L141-L145

Reply:
I'll add something to the v8::Locker implementation that allows us to inspect whether it was necessary or not to acquire the locker.

Reply:
The whole function is gone, what about that? :)